lib/wpc_proto/config.c

```diff
@@ -0,0 +1,682 @@
+/* Copyright 2019 Wirepas Ltd licensed under Apache License, Version 2.0
+ *
+ * See file LICENSE for full license details.
+ *
+ */
+#include <stdio.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <string.h>
+
+#include "platform.h"
+#include "wpc_types.h"
+#include "config.h"
+#include "config_message.pb.h"
+
+#define LOG_MODULE_NAME "Config"
+#define MAX_LOG_LEVEL INFO_LOG_LEVEL
+#include "logger.h"
+
+#include "config_message.pb.h"
+
+/* Set the max of strings used (including \0), matching values defined in proto
+ * files */
+#define GATEWAY_ID_MAX_SIZE      16
+#define SINK_ID_MAX_SIZE         16
+
+/** Structure to hold config from node */
+typedef struct sink_config
+{
+    /* Read only parameters backup-ed with a table (Read at boot up) */
+    uint16_t stack_profile;
+    uint16_t hw_magic;
+    uint16_t ac_limit_min;
+    uint16_t ac_limit_max;
+    uint16_t app_config_max_size;
+    uint16_t version[4];
+    uint8_t  max_mtu;
+    uint8_t  ch_range_min;
+    uint8_t  ch_range_max;
+    uint8_t  pdu_buffer_size;
+
+    /* Read parameters with node interrogation */
+    //uint16_t CurrentAC;
+    bool     CipherKeySet;
+    bool     AuthenticationKeySet;
+    uint8_t  StackStatus;
+
+    /* Read write also ? */
+    uint16_t ac_range_min_cur;  // 0 means unset ?
+    uint16_t ac_range_max_cur;
+
+    /* Read/Write parameters with node interrogation */
+    app_addr_t    node_address;
+    wp_NodeRole   node_role;
+    net_addr_t    network_address;
+    net_channel_t network_channel;
+    // uint32_t channel_map; // deprecated
+
+    /* Methods related to config */
+    msap_app_config_data_write_req_pl_t app_config;
+} sink_config_t;
+
+/* TODO : check how to get access to config cache. Protected access ? */
+/** Sink values read at init time */
+sink_config_t m_sink_config;
+
+static char     m_gateway_id[GATEWAY_ID_MAX_SIZE];
+static char     m_sink_id[SINK_ID_MAX_SIZE];
+static uint32_t m_event_id = 0;
+
+#if 0 // temporary deactivate of read/write access
+
+/**
+ * \brief   Global function to set a key (cipher or authen)
+ * \param   value
+ *          The message received on dbus containing the key
+ * \param   key_set_function
+ *          The WPC key function to use
+ * \return  Return code of operation
+ */
+static app_res_e set_key(sd_bus_message * value, set_key_f key_set_function)
+{
+    const void * key;
+    app_res_e res = APP_RES_INTERNAL_ERROR;
+    size_t n;
+    int r;
+
+    r = sd_bus_message_read_array(value, 'y', &key, &n);
+    if ((r < 0) || (n != 16))
+    {
+        LOGE("Cannot get key from request %s len=%d\n", strerror(-r), n);
+        return res;
+    }
+
+    res = (key_set_function)((uint8_t *) key);
+    return res;
+}
+
+/**
+ * \brief   Set cipher key handler
+ * \param   ... (from sd_bus property handler)
+ */
+static int set_cipher_key(sd_bus * bus,
```

Comment:
Please remove all this code, it has reference to dbus

Reply:
Done